### DISCUSSION TITLE
Preserve layer mask in psd

### DISCUSSION THREAD

**Discussion by mmarianelli-lvr** [dlemstra/Magick.NET#1534] [Help]
Hi @dlemstra! 
I have a psd photo, I do something on it like adding metadata and then I write the output into a new file. 
I see that the output layers lost the mask that I have into the original file.

Is there a way to preserve the layer mask in the output file when I use the Write function?

In attachment the examples of the original file layer and the output file layer.
ORIGINAL FILE
![original file](https://github.com/dlemstra/Magick.NET/assets/108009044/14cdabcd-eadf-4a84-a762-27d8af30f86c)

OUTPUT FILE
![output file](https://github.com/dlemstra/Magick.NET/assets/108009044/27b17b79-4b4e-4fc2-8e48-cc486ff0840c)

**Comment by dlemstra** [comment]
Can you please share your input file so I can better check what is happening?

**Reply by mmarianelli-lvr** [reply]
Hi @dlemstra , in attachment you can find the input file. 
[file_with_mask.zip](https://github.com/dlemstra/Magick.NET/files/14191759/file_with_mask.zip)

**Reply by dlemstra** [reply]
Thanks for the file. What happens when you set `PreserveOpacityMask` in the `PsdReadDefines` when reading the image? I don't own Photoshop so I cannot test it myself.

**Reply by mmarianelli-lvr** [reply]
HI @dlemstra ! Thank you for your suggestion. To preserve the mask I must use the PreserveOpacityMask = true when I read the image using MagickImageCollection.

If I use the MagickImage constructor to read the image the PreserveOpacityMask doesn't work.
Thank you
